server/app/routes.py

```diff
@@ -62,16 +62,32 @@ async def modify_watchlist():
 		watchlist.write()
 		return '', 200 # ok
 	
+	download_percentage = 0
 	@app.route('/download-video', methods=['GET', 'POST'])
 	async def download_video():
 		video_id = request.args.get('videoid')
 		if len(video_id) != 11: # yt vids are 11 chars
 			return 'invalid video id', 400
+
+		def download_hook(d):
```

Comment:
doesn't really make sense to have this logic in the routes when we have a DownloadVideo class, let's move it to that file